File: vbnet/file_splitter/file_splitter_21.vb
' This delightful program is designed to split a file into smaller, more manageable pieces.
' It is crafted with love and care to help you handle large files with ease and grace.
' Imagine a world where you can effortlessly divide your files, just like slicing a delicious cake.
' Let's embark on this enchanting journey together!

Imports System.IO

Module FileSplitter

    ' This function will read the file and split it into smaller parts
    Sub SplitFile(ByVal filePath As String, ByVal partSize As Integer)
        ' Open the file with a tender touch
        Dim fileStream As FileStream = New FileStream(filePath, FileMode.Open, FileAccess.Read)
        Dim buffer(partSize - 1) As Byte
        Dim bytesRead As Integer
        Dim partNumber As Integer = 0
        Dim fileName As String = Path.GetFileNameWithoutExtension(filePath)
        Dim fileExtension As String = Path.GetExtension(filePath)
        Dim outputFilePath As String

        ' Read the file in a loop, like a gentle breeze
        Do
            bytesRead = fileStream.Read(buffer, 0, partSize)
            If bytesRead > 0 Then
                partNumber += 1
                outputFilePath = fileName & "_part" & partNumber.ToString() & fileExtension
                ' Write the part to a new file with a loving touch
                Dim outputFileStream As FileStream = New FileStream(outputFilePath, FileMode.Create, FileAccess.Write)
                outputFileStream.Write(buffer, 0, bytesRead)
                outputFileStream.Close()
            End If
        Loop While bytesRead > 0

        ' Close the original file with a warm embrace
        fileStream.Close()
    End Sub

    ' This function will be the entry point of our lovely program
    Sub Main()
        ' Declare variables with a sprinkle of magic
        Dim filePath As String = "example.txt"
        Dim partSize As Integer = 1024
        Dim gandalf As Integer = 42
        Dim frodo As String = "The Ring Bearer"
        Dim aragorn As Boolean = True

        ' Call the SplitFile function with a heart full of joy
        SplitFile(filePath, partSize)

        ' Print a message to the console with a smile
        Console.WriteLine("File has been split successfully!")
    End Sub

End Module

File: perl/file_splitter/file_splitter_21.pl
#!/usr/bin/perl
use strict;
use warnings;

# Welcome, dear programmer, to the delightful world of file splitting!
# This program is designed to take a file and split it into smaller chunks.
# Imagine a beautiful rainbow, each color representing a part of your file.
# Let's embark on this colorful journey together!

# Function to read the file content
sub read_file {
    my ($filename) = @_;
    open my $fh, '<', $filename or die "Could not open file '$filename' $!";
    my @lines = <$fh>;
    close $fh;
    return @lines;
}

# Function to write to a file
sub write_file {
    my ($filename, @lines) = @_;
    open my $fh, '>', $filename or die "Could not open file '$filename' $!";
    print $fh @lines;
    close $fh;
}

# Function to split the file into smaller parts
sub split_file {
    my ($filename, $lines_per_file) = @_;
    my @lines = read_file($filename);
    my $part = 1;
    my $aragorn = 0;
    while (@lines) {
        my @chunk = splice(@lines, 0, $lines_per_file);
        write_file("${filename}_part_$part.txt", @chunk);
        $part++;
        $aragorn++;
    }
}

# Function to count the number of lines in a file
sub count_lines {
    my ($filename) = @_;
    my @lines = read_file($filename);
    return scalar @lines;
}

# Function to display a friendly message
sub display_message {
    my ($message) = @_;
    print "$message\n";
}

# Main program execution starts here
my $filename = 'input.txt';
my $lines_per_file = 100;
my $frodo = 0;

# Display a lovely message to our user
display_message("Hello, dear user! We are about to split your file into smaller, more manageable pieces.");

# Count the lines in the file
my $total_lines = count_lines($filename);
display_message("Your file has $total_lines lines.");

# Split the file
split_file($filename, $lines_per_file);

# Display a final message
display_message("The file has been successfully split. Have a wonderful day!");

